web/src/beta/components/DragAndDropList/index.tsx

```diff
@@ -25,25 +25,24 @@ function DragAndDropList<Item extends { id: string } = { id: string }>({
   gap,
 }: Props<Item>) {
   const [movingItems, setMovingItems] = useState<Item[]>(items);
-  const listRef = useRef<null | HTMLDivElement>(null);
-
-  const { updatePosition } = useScroll(listRef);
-
+  const { scrollContainerRef } = useScroll();
   const dragDropManager = useDragDropManager();
   const monitor = dragDropManager.getMonitor();
+  const [, setIsMonitor] = useState(0);
```

Comment:
Is this basically acting as a way to rerender the component? 
if so, can you make it clearer somehow through the name of the setter function or wrapping the setIsMonitor in a callback that is named like `handleForceRender` or something?
If I am mistaken, can you explain its purpose? 🙏🏻

Reply:
this is not causing the re-render, the monitor.subscribeToOffsetChange() is used to update the scroll position of the list when an item is being dragged, when the item is dragged, the offset changes, and the preview is updated accordingly. So if not using this subscription, the monitor might not be updated as expected because the subscription to offset changes is no longer in place after the first drag-and-drop operation. This could result in the preview(used as ref item when using custom-dnd-handler) not being updated correctly after the first drag-and-drop operation.